pcsx2/CDVD/CDVD.cpp

```diff
@@ -416,9 +417,14 @@ static bool cdvdUncheckedLoadDiscElf(ElfObject* elfo, IsoReader& isor, const std
 
 bool cdvdLoadElf(ElfObject* elfo, const std::string_view& elfpath, bool isPSXElf, Error* error)
 {
-	if (elfpath.starts_with("host:"))
+	if (R3000A::ioman::is_host(elfpath))
 	{
-		std::string host_filename(elfpath.substr(5));
+		std::string host_filename(elfpath.substr(elfpath.find(':') + 1));
+
+		if (elfo->OpenFile(host_filename, isPSXElf, error))
```

Comment:
Is this still needed? Does `host_path()` below not handle this case?

Reply:
> Does `host_path()` below not handle this case?

Unless I'm missing something, I don't believe it does. `host_path` will transform the path by prefixing it with the host root (i.e., the directory containing the running .elf, .iso, etc.) so a single `OpenFile` alone using `host_path` won't retain the existing behavior.

Reply:
I'm questioning it because the actual HLE open call goes through `host_path()`, so the ELF lookup for PCSX2's internal use and the emulated open are behaving differently, and they probably shouldn't.

`host_path()` should also preserve the path if it's absolute.

Reply:
`host_path` currently deliberately disallows the use of absolute paths, do I understand correctly that you are proposing to change that and remove the restriction? If not, I suppose an alternative would be to do this instead of the two `OpenFile` calls?:
```cpp
std::string host_filename(elfpath.substr(elfpath.find(':') + 1));

if (!Path::IsAbsolute(host_filename))
    host_filename = R3000A::ioman::host_path(host_filename, false);

return elfo->OpenFile(host_filename, isPSXElf, error);
```

Reply:
I'm suggesting the behaviour should be the same as the HLE methods, e.g. open():
https://github.com/PCSX2/pcsx2/blob/3114332c8999068a9a88e6aceae3f33ecb5a437b/pcsx2/IopBios.cpp#L303-L306

Reply:
I just tested with the following implementation, which should match the HLE methods:
```cpp
if (R3000A::ioman::is_host(elfpath))
{
    const std::string_view path(elfpath.substr(elfpath.find(':') + 1));
    const std::string file_path(R3000A::ioman::host_path(path, false));
    return elfo->OpenFile(file_path, isPSXElf, error);
}
```
Everything continues to work as it should, to my surprise. The source of my confusion was me not realizing that the host root is empty when this code runs for the initial ELF, so absolute paths are allowed through. I'll change it to use this version now.